controllers/search.js

```diff
@@ -0,0 +1,175 @@
+import Sequelize from 'sequelize';
+import models from '../models';
+
+const { User, article } = models;
+
+const userAttrib = ['username', 'image', 'bio'];
+const articleAttrib = ['title', 'description', 'slug', 'tagList'];
+
+const handler = {
+  searchByAuthor: async (req, res) => {
+    const { title, username } = req.body;
+
+    if (!title && !username) {
+      return res.status(400).send({
+        status: res.statusCode,
+        error: 'Please search using the article title or authors\' username'
+      });
+    }
+
+    try {
+      /**
+      * Any user is able to search for an article if
+      * they know it's tittle
+      */
+
+      if (title) {
+        const articles = await article.findAll(
+          {
+            limit: 10,
+            attributes: articleAttrib,
+            where: {
+              title: { [Sequelize.Op.iLike]: `%${title}%` }
+            },
+            include: {
+              model: User,
+              attributes: userAttrib,
+            }
+          }
+        );
+        return res.status(201).send({
+          status: res.statusCode,
+          articles
+        });
+      }
+
+      /**
+      * Any user is able to search for an article if
+      * they know the authors name
+      */
+
+      if (username) {
+        const users = await User.findAll({
+          where: {
+            username: { [Sequelize.Op.iLike]: `%${username}%` }
+          },
+          attributes: userAttrib,
+          include: [{
+            model: article,
+            attributes: articleAttrib,
+          }],
+          limit: 10,
+        });
+        return res.status(201).send({
+          status: res.statusCode,
+          users
+        });
+      }
+    } catch (error) {
+      return res.status(500).send({
+        status: res.statusCode,
+        error: 'Server failed to handle your request'
+
+      });
+    }
+  },
+
+  searchByBody: async (req, res) => {
+    const { body } = req.body;
+    try {
+      const articles = await article.findAll({
+        where: {
+          body: {
+            [Sequelize.Op.iLike]: `%${body}%`
```

Comment:
Consider trimming this variable too! a single space body will return all articles!

Reply:
Fixed